src/dos/drive_local.cpp

```diff
@@ -438,11 +475,12 @@ bool localDrive::GetFileAttr(char *name, uint16_t *attr)
 #else
 	struct stat status;
 	if (stat(newname, &status) == 0) {
-		*attr = status.st_mode & S_IFDIR ? 0 : DOS_ATTR_ARCHIVE;
-		if (status.st_mode & S_IFDIR)
-			*attr |= DOS_ATTR_DIRECTORY;
-		if (!(status.st_mode & S_IWUSR))
-			*attr |= DOS_ATTR_READ_ONLY;
+		// HACK: Always report Archive attribute on Unix since managing it
+		// manually would be a pain.
```

Comment:
How so? We should add support for all attributes, including archive. How is this more difficult than supporting the rest of the FAT attributes?

Reply:
Because Archive isn't just an attribute, it's automatically added by OS whenever the file is modified.
https://en.wikipedia.org/wiki/Design_of_the_FAT_file_system#attributes

Reply:
That Wikipedia page states this:

> [Archive](https://en.wikipedia.org/wiki/Archive_bit). (Since DOS 2.0) Typically set by the operating system as soon as the file is created or modified to mark the file as "dirty", and reset by backup software once the file has been backed up to indicate "pure" state.

So you simply set the flag in the create file and write file calls in the DOSBox code, I don't see the difficulty. Surely we don't have hundreds of functions that needs this adjustment, but only a few. Certainly doable with minimal extra effort 😄 

Then the modify attribute call, and by extension the ATTRIB command could clear (or set) this flag.

If you go to the trouble of adding attribute support, then let's do a complete job and do it properly.